src/server/authMiddleware.ts

```diff
@@ -46,7 +55,19 @@ const isPublicRoute = (req: NextRequest, options: MiddlewareOptions) => {
 	);
 	const isPublic = options.publicRoutes?.includes(req.nextUrl.pathname);
 
-	return isDefaultPublicRoute || isPublic;
+	// If both publicRoutes and privateRoutes are provided, we prioritize publicRoutes
+	if (options.publicRoutes && options.privateRoutes) {
+		return isDefaultPublicRoute || isPublic;
+	}
+
+	// If only publicRoutes are provided
+	if (options.publicRoutes) {
+		return isDefaultPublicRoute || isPublic;
+	}
+
+	// If only privateRoutes are provided
+	const isPrivate = options.privateRoutes?.includes(req.nextUrl.pathname);
+	return isDefaultPublicRoute || !isPrivate;
```

Comment:
few things about this implemnation 

1. need to handle empty array like above
2. there is duplication in conditions in lines 59 and 64, the second condition is a subset of the first, making the first redundant